### PR TITLE
feat: Google Analytics 4 middleware

### PR DESCRIPTION
**Pull request by roj1512**

### CODE REVIEW

File: testdata/my_blog.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/** @jsx h */
```
Comment (bartlomieju) [inline testdata/my_blog.tsx:1]: Unrelated file

File: deps.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export {
   createReporter,
   type Reporter as GaReporter,
 } from "https://deno.land/x/g_a@0.1.2/mod.ts";
+export { GA4Report } from "https://raw.githubusercontent.com/denoland/ga4/main/mod.ts";
```
Comment (bartlomieju) [inline deps.ts:28]: We should wait until this library is officialy released before landing this PR

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: testdata/my_blog.tsx (hunk 1 above)
Comment (roj1512) [reply testdata/my_blog.tsx:1]: my bad

File: deps.ts (hunk 2 above)
Comment (roj1512) [reply deps.ts:28]: alright

Comment (roj1512) [reply deps.ts:28]: leaving this convo unresolved until the google analytics 4 experiments are done your side :)

**Comment by roj1512** [comment]
Marking as draft in favor of https://github.com/denoland/deno_blog/pull/31#discussion_r907651268

**Comment by KorigamiK** [comment]
This was never merged, are there any updates since Universal Analytics is going to shut down soon.

**Comment by roj1512** [comment]
CC @bartlomieju
